src/plugins/MediaManager.js

```diff
@@ -0,0 +1,341 @@
+        }
+        this.instance.renegotiateWebRTCConnection();
+    }
+
+    /**
+     * Add a local video stream and send it through SDP renegotiation.
+     *
+     * See https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/addTrack
+     * and https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/addStream
+     *
+     * @param {MediaStream} stream Video stream to add.
+     */
+    addVideoStream(stream) {
+        if (!this.instance.peerConnection) {
+            log.error('Could not add video stream: connection is not ready');
+            return;
+        }
+        /**
+         * If cameraSender is defined, this means that we already added
+         * it to the PeerConnection. Since removeTrack() just remove the track
+         * from it, re-add it and switch back the RTCRtpTranceiver direction
+         * to send and recv.
+         */
+        if (stream.getVideoTracks().length > 0) {
+            if (this.cameraSender) {
+                log.debug('Replacing video track on sender');
+                this.cameraSender.replaceTrack(stream.getVideoTracks()[0]);
+                this.setTransceiverDirection(this.cameraSender, 'sendrecv');
+            } else {
+                this.cameraSender = this.instance.peerConnection.addTrack(stream.getVideoTracks()[0],
+                    stream);
+            }
+        }
+
+        if (this.videoWithMicrophone && stream.getAudioTracks().length > 0) {
+            if (this.microphoneSender) {
+                log.debug('Replacing audio track on sender');
+                this.microphoneSender.replaceTrack(stream.getAudioTracks()[0]);
+                this.setTransceiverDirection(this.microphoneSender, 'sendrecv');
+            } else {
+                this.microphoneSender = this.instance.peerConnection.addTrack(stream.getAudioTracks()[0],
+                    stream);
+            }
+        }
+        this.instance.renegotiateWebRTCConnection();
+    }
+
+    /**
+     * Remove a local audio stream and stop sending it through SDP renegotiation.
+     *
+     * See https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/removeTrack
+     * and https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/removeStream
+     *
+     * @param {MediaStream} stream Audio stream to stop and remove.
+     */
+    removeAudioStream(stream) {
+        if (!this.instance.peerConnection) {
+            log.error('Could not remove audio stream: connection is not ready');
+            return;
+        }
+        if (stream instanceof MediaStream) {
+            stream.getTracks().forEach((track) => {
+                track.stop();
+            });
+            if (this.microphoneSender) {
+                this.instance.peerConnection.removeTrack(this.microphoneSender);
+            }
+            this.instance.renegotiateWebRTCConnection();
+        }
+    }
+
+    /**
+     * Remove a local video stream and stop sending it through SDP renegotiation.
```

Comment:
It also remove the audio track not just the video. Not sure if it is really intended now we split audio and video

Reply:
I should add comments to further explain that.